rust/otel-arrow-rust/src/otap/batching.rs

```diff
@@ -0,0 +1,259 @@
+//! Batching for `OtapArrowRecords`
+//!
+//!
+
+use std::{num::NonZeroU64, time::Duration};
+
+use tokio::{
+    sync::{
+        mpsc::{UnboundedReceiver, UnboundedSender, unbounded_channel},
+        oneshot,
+    },
+    task::JoinHandle,
+    time::{Instant, sleep},
+};
+
+use super::{OtapArrowRecords, error::Result, groups::RecordsGroup};
+
+/// Control how we batch data
+///
+/// Specifically, how much data can we accumulate before we emit a batch, how long can we store that
+/// data before emitting a batch, and what's the largest batch we'll emit.
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub struct Config {
+    /// Equivalent to `send_batch_size` in the go implementation.
+    size_trigger: NonZeroU64,
+
+    /// Equivalent to `timeout` in the go implementation.
+    time_trigger: Option<Duration>,
+
+    /// Equivalent to `send_batch_max_size` in the go implementation.
+    max_output_batch: Option<NonZeroU64>,
```

Comment:
I think it’s good to have the mapping with the Go implementation parameters, but it would also be useful to have a local description of what each of these parameters does for those who are not familiar with the Go version.